store/src/sql/access/stream.rs

```diff
@@ -0,0 +1,147 @@
+use anyhow::anyhow;
+use cid::Cid;
+
+use crate::{
+    sql::entities::{
+        EventHeader, EventHeaderRow, EventType, IncompleteStream, StreamCid, StreamCommitRow,
+        StreamRow,
+    },
+    Error, Result, SqlitePool, SqliteTransaction,
+};
+
+/// Access to the stream and related tables. Generally querying events as a stream.
+pub struct CeramicOneStream {}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// Represents a stream event in a way that allows ordering it in the stream. It is metadata and not the event payload itself.
+pub struct StreamCommit {
+    /// The event CID
+    pub cid: Cid,
+    /// The previous event CID
+    pub prev: Option<Cid>,
+    /// Whether the event has been delivered
+    pub delivered: bool,
+}
+
+impl TryFrom<StreamCommitRow> for StreamCommit {
+    type Error = crate::Error;
+
+    fn try_from(row: StreamCommitRow) -> std::result::Result<Self, Self::Error> {
+        let cid = Cid::try_from(row.cid)
+            .map_err(|e| Error::new_app(anyhow!("Invalid event cid: {}", e)))?;
+        let prev = row
+            .prev
+            .map(Cid::try_from)
+            .transpose()
+            .map_err(|e| Error::new_app(anyhow!("Invalid event prev: {}", e)))?;
+        Ok(Self {
+            cid,
+            prev,
+            delivered: row.delivered,
+        })
+    }
+}
+
+impl CeramicOneStream {
+    /// Load the events (commits) for a given stream
+    pub async fn load_stream_commits(
+        pool: &SqlitePool,
+        stream_cid: StreamCid,
+    ) -> Result<Vec<StreamCommit>> {
+        let rows: Vec<(Vec<u8>, Option<Vec<u8>>, bool)> =
+            sqlx::query_as(StreamCommitRow::fetch_by_stream_cid())
+                .bind(stream_cid.to_bytes())
+                .fetch_all(pool.reader())
+                .await?;
+
+        let res = rows
+            .into_iter()
+            .map(|(cid, prev, delivered)| {
+                let cid = Cid::try_from(cid).expect("cid");
+                let prev = prev.map(Cid::try_from).transpose().expect("prev");
+
+                StreamCommit {
+                    cid,
+                    prev,
+                    delivered,
+                }
+            })
+            .collect();
+
+        Ok(res)
+    }
+
+    /// Load streams with undelivered events to see if they need to be delivered now.
+    pub async fn load_stream_cids_with_undelivered_events(
+        pool: &SqlitePool,
+        last_cid: StreamCid,
```

Comment:
Can you add a comment for what is `last_cid`?

Reply:
Yeah, good call. I was going to call it `cid_cursor` or something but wasn't sure how to best convey the "start after this CID" semantic.